src/tpm2/Object.c

```diff
@@ -460,7 +460,10 @@ ObjectLoad(
     object->publicArea = *publicArea;
     // If there is a sensitive area, load it
     if(sensitive == NULL)
-	object->attributes.publicOnly = SET;
+	{		// libtpms changed begin
+	    object->attributes.publicOnly = SET;
+	    MemorySet(&object->sensitive, 0, sizeof(object->sensitive)); // needed for libtpms.
```

Comment:
Actually, are you sure that this is necessary? Do you have a test case? I am asking because `ObjectLoad()` is called with an object pointer from two locations:

```
-> src/tpm2/ObjectCommands.c:    result = ObjectLoad(newObject, parentObject,
-> src/tpm2/ObjectCommands.c:    result = ObjectLoad(object, NULL,
not this one: src/tpm2/DuplicationCommands.c:     result = ObjectLoad(NULL, NULL, &in->objectPublic.publicArea,
```

Both get the object they pass into this function like this:
```
    newObject = FindEmptyObjectSlot(&out->objectHandle);

    object = FindEmptyObjectSlot(&out->objectHandle);
```

FindEmptyObjectSlot has been instrumented previously to clear the object:

```
OBJECT *
FindEmptyObjectSlot(
                    TPMI_DH_OBJECT  *handle         // OUT: (optional)
                    )
{
    UINT32               i;
    OBJECT              *object;
    for(i = 0; i < MAX_LOADED_OBJECTS; i++)
        {
            object = &s_objects[i];
            if(object->attributes.occupied == CLEAR)
                {
                    if(handle)
                        *handle = i + TRANSIENT_FIRST;
                    // Initialize the object attributes
                    // MemorySet(&object->attributes, 0, sizeof(OBJECT_ATTRIBUTES));
                    MemorySet(object, 0, sizeof(*object)); // libtpms added: Initialize the whole object
                    return object;
                }
        }
    return NULL;
}
```

Reply:
Apologies, you are absolutely right. I was using a (much) older version that didn't have 17255da54cf8354d02369f1323dc50cfb87e2bf4 and that addresses the problem I was seeing. This can be reverted.